### PR TITLE
[cpp] improve single quote string highlighting

### PR DESCRIPTION
**Pull request by yyyc514**

### CODE REVIEW

**Comment by sylveon** [comment]
In C++, single quotes are for single character literals, not string literals. This code fixes the issue by treating them as if they where string literals, which is fine, but the example should probably not use them like so.

**Comment by sylveon** [comment]
While we're at it, it doesn't seems like wide character raw strings are handled.
```cpp
LR"(hello
world)"
```
doesn't highlights the L even though it should be.

**Comment by yyyc514** [comment]
Ah, duh, so that explain the weirdness of the existing regex.  Ok, back to the drawing board on this, not read to merge yet. :-)

> doesn't highlights the L even though it should be.

Is it highlighting the rest?

Do you understand the regex for raw strings:

```js
{ begin: /(?:u8?|U|L)?R"([^()\\ ]{0,16})\((?:.|\n)*?\)\1"/ },
```

I'm not sure what we're checking for at the beginning with the {0,16}, do you know?

**Comment by yyyc514** [comment]
> In C++, single quotes are for single character literals, not string literals.

Can a character literal be MORE than one literal character wide then if it's encoded as u8 or U or L?

**Comment by sylveon** [comment]
Even without being Unicode it can, due to being able to create character
literals for escapes, eg '\n'

**Comment by sylveon** [comment]
And you also have unicode character escapes, like `L'\u202e'` or hexadecimal character escapes like `'\xFF'`

**Comment by yyyc514** [comment]
I meant OTHER than escapes. :-)    YES, those other ones are the more interesting ones.

**Comment by sylveon** [comment]
It'd be easier to handle character literals as a normal string, they're the same escapes and even normal strings don't seem to highlight those anyways.

As for raw strings, it does highlight the text, only the `L` isn't. The regex seems to be doing some things to correctly detect custom prefixes for raw strings:
```cpp
LR"foobar(hello
world)foobar"
```
Those custom delimiters can be nothing or at most 16 characters and must not contain parentheses, spaces or backslashes, hence the `([^()\\ ]{0,16})`. See https://en.cppreference.com/w/cpp/language/string_literal

**Comment by sylveon** [comment]
As for L not being highlighted it's probably just because of an out of date version, I used hastebin (which uses highlight.js to highlight its stuff) to test out of laziness, the regex seems to be doing the correct thing now but has no associated test case.

**Comment by yyyc514** [comment]
Better?

File: test/markup/index.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ function testLanguage(language) {
 
           actual.trim().should.equal(expected.trim());
           done();
-        });
+        }).catch(function(err) { return done(err) });
```
Comment (sylveon) [inline test/markup/index.js]: Not sure if this was intended to be left there.

Comment (egor-rogov) [reply test/markup/index.js]: It's already merged in fact.

**Comment by egor-rogov** [comment]
Merged, thanks everybody!

File: test/markup/index.js (hunk 1 above)
Comment (yyyc514) [reply test/markup/index.js]: Yes, this is intentional.  It allows the test runner to properly colorize errors for debugging failing tests.

**Pull request merged**